### PR TITLE
migrate more from Travis to GHA

### PR DESCRIPTION
**Pull request by reaperhulk**

### CODE REVIEW

**Comment by geoffreyblake** [comment]
Hi pyca/cryptograhpy devs, can you contact support@travis-ci.com? They're working on getting the necessary time credits to OSS projects.

**Comment by reaperhulk** [comment]
@geoffreyblake We actually appear to be on a queue exempt from the new rules, but we're not interested in the risk of staying on Travis. Past experience has shown that these sorts of changes only ratchet tighter and needing to repeatedly engage to preserve our resources when they almost inevitably get shut off at the most inconvenient time isn't worth the headache. We can transition to entirely GHA + OpenDev (which provides our AArch64) so that's our plan for now.

This change won't affect our CI capabilities at all, so if you were concerned about loss of AArch64 we're okay there 😄 .

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -10,6 +10,100 @@ on:
...
+              popd
+          fi
+        env:
+          TYPE: ${{ matrix.PYTHON.OPENSSL.TYPE }}
+          VERSION: ${{ matrix.PYTHON.OPENSSL.VERSION }}
+        if: ${{ matrix.PYTHON.OPENSSL }} && steps.cache.outputs.cache-hit != 'true'
+      - name: Set CFLAGS/LDFLAGS
+        run: |
+          echo "CFLAGS=\"${CFLAGS} -I${OSSL_PATH}/include\"" >> $GITHUB_ENV
+          echo "LDFLAGS=\"${LDFLAGS} -L${OSSL_PATH}/lib -Wl,-rpath=${OSSL_PATH}/lib\"" >> $GITHUB_ENV
```
Comment (alex) [inline .github/workflows/ci.yml:98]:
```suggestion
          echo "CFLAGS=${CFLAGS} -I${OSSL_PATH}/include" >> $GITHUB_ENV
          echo "LDFLAGS=${LDFLAGS} -L${OSSL_PATH}/lib -Wl,-rpath=${OSSL_PATH}/lib" >> $GITHUB_ENV
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -10,6 +10,106 @@ on:
...
+          VERSION: ${{ matrix.PYTHON.OPENSSL.VERSION }}
+        if: matrix.PYTHON.OPENSSL && steps.ossl-cache.outputs.cache-hit != 'true'
+      - name: Set CFLAGS/LDFLAGS
+        run: |
+          echo "CFLAGS=${CFLAGS} -I${OSSL_PATH}/include" >> $GITHUB_ENV
+          echo "LDFLAGS=${LDFLAGS} -L${OSSL_PATH}/lib -Wl,-rpath=${OSSL_PATH}/lib" >> $GITHUB_ENV
+        if: matrix.PYTHON.OPENSSL
+      - name: Tests
+        run: |
+            echo "Expected OpenSSL path: ${OSSL_PATH}"
```
Comment (alex) [inline .github/workflows/ci.yml:103]: Can this be removed now?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -10,6 +10,106 @@ on:
...
+            tox -r --  --color=yes --wycheproof-root=wycheproof
+        env:
+          TOXENV: ${{ matrix.PYTHON.TOXENV }}
+      - name: Upload coverage
+        run: |
+          curl -o codecov.sh -f https://codecov.io/bash || curl -o codecov.sh -f https://codecov.io/bash || curl -o codecov.sh -f https://codecov.io/bash
+          bash codecov.sh -n "tox -e ${{ matrix.PYTHON.TOXENV }} ${{ matrix.PYTHON.OPENSSL }}"
+        if: matrix.PYTHON.COVERAGE == 'true'
```
Comment (alex) [inline .github/workflows/ci.yml:111]: Can we make this `if != "false"` so we don't need the key in every single element?

**Review by alex — COMMENTED** [review]
You should be able to delete a ton of code from `.travis` now, right?

**Review by alex — APPROVED** [review]

**Review by alex — APPROVED** [review]

**Pull request merged**